Marlin/src/pins/stm32f4/pins_CREALITY_CR4NTxxC10.h

```diff
@@ -0,0 +1,258 @@
+  #define Z_MIN_PIN                         PB0   // BLTouch IN PIN (TOUCH pins in the schematic have changed)
+  #define Z_MAX_PIN                         PA7
+#else
+  #define Z_STOP_PIN                        PA7   // Z-axis limit switch
+#endif
+
+#ifndef Z_MIN_PROBE_PIN
+  #define Z_MIN_PROBE_PIN                   PB0   // BLTouch IN
+#endif
+
+//
+// Filament Runout Sensor
+//
+#ifndef FIL_RUNOUT_PIN
+  #define FIL_RUNOUT_PIN                    PA4   // "Pulled-high"
+#endif
+
+//
+// Steppers
+//
+#define X_ENABLE_PIN                        PC3
+#define X_STEP_PIN                          PC2
+#define X_DIR_PIN                           PB9
+
+#define Y_ENABLE_PIN                        PC3
+#define Y_STEP_PIN                          PB8
+#define Y_DIR_PIN                           PB7
+
+#define Z_ENABLE_PIN                        PC3
+#define Z_STEP_PIN                          PB6
+#define Z_DIR_PIN                           PB5
+
+#define E0_ENABLE_PIN                       PC3
+#define E0_STEP_PIN                         PB4
+#define E0_DIR_PIN                          PB3
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN                          PC5   // TH1
+#define TEMP_BED_PIN                        PC4   // TB1
+
+//
+// Heaters / Fans
+//
+#define HEATER_0_PIN                        PA1   // HEATER1
+#define HEATER_BED_PIN                      PB10  // HOT BED
+
+#ifndef FAN0_PIN
+  #define FAN0_PIN                          PA0   // FAN
+#endif
+#if PIN_EXISTS(FAN0)
+  #define FAN_SOFT_PWM
+#endif
+
+#if HAS_TMC_UART
+  // Reduce baud rate to improve software serial reliability
+  #define TMC_BAUD_RATE 19200
+
+  /**
+   * TMC2208/TMC2209 stepper drivers
+   *
+   * Hardware serial communication ports.
+   * If undefined software serial is used according to the pins below
+   */
+  #define X_HARDWARE_SERIAL  Serial6
+  #define Y_HARDWARE_SERIAL  Serial6
+  #define Z_HARDWARE_SERIAL  Serial6
+  #define E0_HARDWARE_SERIAL Serial6
+
+  #ifndef X_SLAVE_ADDRESS
+    #define X_SLAVE_ADDRESS 0
+  #endif
+  #ifndef Y_SLAVE_ADDRESS
+    #define Y_SLAVE_ADDRESS 1
+  #endif
+  #ifndef Z_SLAVE_ADDRESS
+    #define Z_SLAVE_ADDRESS 2
+  #endif
+  #ifndef E0_SLAVE_ADDRESS
+    #define E0_SLAVE_ADDRESS 3
+  #endif
+
+  // Software serial
+
+  #define X_SERIAL_TX_PIN                   PC6
+  #define X_SERIAL_RX_PIN                   PC7
+
+  #define Y_SERIAL_TX_PIN        X_SERIAL_TX_PIN
+  #define Y_SERIAL_RX_PIN        X_SERIAL_RX_PIN
+
+  #define Z_SERIAL_TX_PIN        X_SERIAL_TX_PIN
+  #define Z_SERIAL_RX_PIN        X_SERIAL_RX_PIN
+
+  #define E0_SERIAL_TX_PIN       X_SERIAL_TX_PIN
+  #define E0_SERIAL_RX_PIN       X_SERIAL_RX_PIN
+
+  #define E1_SERIAL_TX_PIN       X_SERIAL_TX_PIN
+  #define E1_SERIAL_RX_PIN       X_SERIAL_RX_PIN
+
+  #define X_DIAG_PIN                        PC13
+  #define Y_DIAG_PIN                        PC14
+  #define Z_DIAG_PIN                        PC15
+  #define E0_DIAG_PIN                       PA15
```

Comment:
If this board has dedicated DIAG pins they won't depend on TMC UART, but should be available with any compatible TMC driver. I presume the drivers are soldered on the board, so these pins are probably always available, even if they change to SPI-based or `_STANDALONE` drivers.

Reply:
Yes, the drivers are soldered onto the board.

I couldn't find a schematic nor a pinout for this board so I have no idea if there are jumpers for the DIAG pins.  

I think this is change is correct.